LayerSelector.js

```diff
@@ -725,6 +725,24 @@ define([
             array.forEach(this.overlayWidgets, startup);
 
             this.inherited(arguments);
+        },
+        /**
+         * Remove layers from the map before destroying the widget.
+         */
+        destroy: function () {
+            console.log('layer-selector:destroy', arguments);
+
+            var managedLayers = this.get('managedLayers');
+            for (var layerName in managedLayers) {
+                if (managedLayers.hasOwnProperty(layerName)) {
```

Comment:
I would maybe try to reduce the arrow functionness of this in the future and consider object.keys but it works for me.